addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1663,7 +1663,8 @@ const Wysiwyg = Widget.extend({
         if (color && (!ColorpickerWidget.isCSSColor(color) && !weUtils.isColorGradient(color))) {
             color = (eventName === "foreColor" ? 'text-' : 'bg-') + color;
         }
-        const coloredElements = this.odooEditor.execCommand('applyColor', color, eventName === 'foreColor' ? 'color' : 'backgroundColor', this.lastMediaClicked);
+        let coloredElements = this.odooEditor.execCommand('applyColor', color, eventName === 'foreColor' ? 'color' : 'backgroundColor', this.lastMediaClicked);
+        coloredElements = coloredElements.filter(element => this.odooEditor.document.contains(element))
```

Comment:
Filtering is the right fix but it seems out of place. Any time we call `execCommand('applyColor', ...` this will return elements that are not in the dom anymore, so in theory, the fix should be called while calling `execCommand('applyColor', ...`. The problem is that the sanitization is called by `historyStep`, so it is completely ad-hoc to add the fix there. 
@dmo-odoo What do you think? Should we let the fix be local or we add some mechanism to fix it "in the right place"?
In practice, the returned nodes of `execCommand('applyColor'` are only used here once (in `_processAndApplyColor`). So a more general fix is only usefull for a future usage of `execCommand('applyColor'` that needs to manipulate the nodes.

But in any case, I would add a comment on top of the line fix to explain the reason (that some node returned by the applyColor command can be removed by the `sanitization` happening in `historyStep`)

Also, there is a missing semicolon at the end of the line :innocent:

Reply:
I agree it seems wrong to have to filter the nodes that were removed by the sanitizer. Theoretically, `applyRawCommand` could be called to be able to do manipulations after and then call `historyStep` manually. That would be an easy workaround. But in the end it seems to call for a refactor so that this can be done in the command itself (maybe an option?). In any case that's clearly out of scope so I suggest either keeping the fix as is but with a very clear comment, or using my other suggestion, and adding this to the list of refactoring tasks for us.

Reply:
@Goaman I added the comment you requested (and the semicolon)